src/Core/Resolvers/MsSqlQueryBuilder.cs

```diff
@@ -63,27 +65,147 @@ public string Build(SqlQueryStructure structure)
+            bool isInsertDMLTriggerEnabled = sourceDefinition.IsInsertDMLTriggerEnabled;
+            string tableName = $"{QuoteIdentifier(structure.DatabaseObject.SchemaName)}.{QuoteIdentifier(structure.DatabaseObject.Name)}";
+
+            // Predicates by virtue of database policy for Create action.
+            string dbPolicypredicates = JoinPredicateStrings(structure.GetDbPolicyForOperation(EntityActionOperation.Create));
+
+            // Columns whose values are provided in the request body - to be inserted into the record.
             string insertColumns = Build(structure.InsertColumns);
-            string insertIntoStatementPrefix = $"INSERT INTO {QuoteIdentifier(structure.DatabaseObject.SchemaName)}.{QuoteIdentifier(structure.DatabaseObject.Name)} ({insertColumns}) " +
-                $"OUTPUT {MakeOutputColumns(structure.OutputColumns, OutputQualifier.Inserted)} ";
-            string values = predicates.Equals(BASE_PREDICATE) ?
-                $"VALUES ({string.Join(", ", structure.Values)});" : $"SELECT {insertColumns} FROM (VALUES({string.Join(", ", structure.Values)})) T({insertColumns}) WHERE {predicates};";
-            StringBuilder insertQuery = new(insertIntoStatementPrefix);
-            return insertQuery.Append(values).ToString();
+
+            // Values to be inserted into the entity.
+            string values = dbPolicypredicates.Equals(BASE_PREDICATE) ?
+                $"VALUES ({string.Join(", ", structure.Values)});" : $"SELECT {insertColumns} FROM (VALUES({string.Join(", ", structure.Values)})) T({insertColumns}) WHERE {dbPolicypredicates};";
+
+            // Final insert query to be executed against the database.
+            StringBuilder insertQuery = new();
+            if (!isInsertDMLTriggerEnabled)
+            {
+                // When there is no DML trigger enabled on the table for insert operation, we can use OUTPUT clause to return the data.
+                insertQuery.Append($"INSERT INTO {tableName} ({insertColumns}) OUTPUT " +
+                    $"{MakeOutputColumns(structure.OutputColumns, OutputQualifier.Inserted.ToString())} ");
+                insertQuery.Append(values);
+            }
+            else
+            {
+                // When a DML trigger for INSERT operation is enabled on the table, its a bit tricky to get the inserted data.
+                // We need to insert the values for all the non-autogenerated columns in the PK into a temporary table.
+                // Finally this temporary table will be used to do a subsequent select on the actual table where we would join the
+                // actual table and the temporary table based on the values of the non-autogenerated PK columns.
+                // If there is a column in the PK which is autogenerated, we cannot and will not insert it into the temporary table.
+                // Hence in the select query, we will add an additional WHERE predicate (alongwith the join).
```

Comment:
additional WHERE predicate for the SELECT on the actual table? that would be a helpful clarification